presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationJdbcSmokeTest.java

```diff
@@ -0,0 +1,111 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.testing.MaterializedResult;
+import org.testng.annotations.Test;
+
+import java.sql.SQLException;
+
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public abstract class AbstractTestIntegrationJdbcSmokeTest
+        extends AbstractTestIntegrationSmokeTest
+{
+    protected AbstractTestIntegrationJdbcSmokeTest(QueryRunnerSupplier supplier)
+    {
+        super(supplier);
+    }
+
+    @Test
+    public void testShowCreateTable()
+            throws SQLException
+    {
+        try {
+            String catalog = getSession().getCatalog().get();
+            assertUpdate("CREATE TABLE test_show_create (" +
+                    "column_a DATE," +
+                    "column_b DATE NOT NULL" +
+                    ")");
+            MaterializedResult materializedRows = computeActual("SHOW CREATE TABLE tpch.test_show_create");
+            assertNotNull(materializedRows);
+            assertEquals(materializedRows.getMaterializedRows().get(0).getFields().get(0),
+                    "CREATE TABLE " + catalog + ".tpch.test_show_create (\n" +
+                            "   column_a date,\n" +
+                            "   column_b date NOT NULL\n" +
+                            ")");
+        }
+        finally {
+            assertUpdate("DROP TABLE tpch.test_show_create");
+        }
+    }
+
+    @Test
+    public void testShowCreateTableAndInsertWithNonNull()
+            throws SQLException
+    {
+        try {
+            assertUpdate("CREATE TABLE test_not_null_with_insert (" +
+                    "column_a DATE," +
+                    "column_b DATE NOT NULL" +
+                    ")");
+
+            assertUpdate("INSERT INTO tpch.test_not_null_with_insert(column_b) VALUES (date '2012-12-31')", 1);
+            assertUpdate("INSERT INTO tpch.test_not_null_with_insert(column_a, column_b) VALUES (date '2013-01-01', date '2013-01-02')", 1);
+            assertQuery(
+                    "SELECT * FROM tpch.test_not_null_with_insert",
+                    "SELECT NULL, CAST ('2012-12-31' AS DATE) UNION ALL SELECT CAST ('2013-01-01' AS DATE), CAST ('2013-01-02' AS DATE);");
+        }
+        finally {
+            assertUpdate("DROP TABLE tpch.test_not_null_with_insert");
+        }
+    }
+
+    @Test(expectedExceptions = RuntimeException.class)
```

Comment:
This will also catch the exception that assert failures? Add a expectedExceptionsMessageRegExp.